apps/health_campaign_field_worker_app/lib/pages/beneficiary/beneficiary_details.dart

```diff
@@ -0,0 +1,407 @@
+              ? taskData.last.additionalFields?.fields
+                      .firstWhereOrNull((e) =>
+                          e.key == AdditionalFieldsType.doseIndex.toValue())
+                      ?.value ??
+                  '1'
+              : '0';
+          final lastCycle = taskData != null && taskData.isNotEmpty
+              ? taskData.last.additionalFields?.fields
+                      .firstWhereOrNull((e) =>
+                          e.key == AdditionalFieldsType.cycleIndex.toValue())
+                      ?.value ??
+                  '1'
+              : '1';
+
+          // [TODO] Need to move this to Bloc Lisitner or consumer
+          if (projectState.projectType != null) {
+            bloc.add(
+              DeliverInterventionEvent.setActiveCycleDose(
+                taskData != null && taskData.isNotEmpty
+                    ? int.tryParse(
+                          lastDose,
+                        ) ??
+                        1
+                    : 0,
+                taskData != null && taskData.isNotEmpty
+                    ? int.tryParse(
+                          lastCycle,
+                        ) ??
+                        1
+                    : 1,
+                state.selectedIndividual,
+                projectState.projectType!,
+              ),
+            );
+          }
+
+          // Building the table content based on the DeliverInterventionState
+
+          return BlocBuilder<ProductVariantBloc, ProductVariantState>(
+            builder: (context, productState) {
+              return productState.maybeWhen(
+                orElse: () => const Offstage(),
+                fetched: (productVariantsvalue) {
+                  final variant = productState.whenOrNull(
+                    fetched: (productVariants) {
+                      return productVariants;
+                    },
+                  );
+
+                  return Scaffold(
+                    body: ReactiveFormBuilder(
+                      form: buildForm,
+                      builder: (context, formGroup, child) => ScrollableContent(
+                        header: const Column(children: [
+                          BackNavigationHelpHeaderWidget(),
+                        ]),
+                        footer: BlocBuilder<DeliverInterventionBloc,
+                            DeliverInterventionState>(
+                          builder: (context, deliverState) {
+                            final projectType = projectState.projectType;
+                            final cycles = projectType?.cycles;
+
+                            return cycles != null && cycles.isNotEmpty
+                                ? deliverState.hasCycleArrived
+                                    ? SizedBox(
+                                        height: 100,
+                                        child: DigitCard(
+                                          margin:
```

Comment:
why have hardcoded the size

Reply:
To have a fixed size for the button, for all type of screens